Firestore/core/test/firebase/firestore/local/lru_garbage_collector_test.cc

```diff
@@ -0,0 +1,740 @@
+    UpdateTargetInTransaction(middle_target);
+  });
+
+  // middle_target removed here, no update needed
+
+  // Write a doc and get an ack, not part of a target.
+  persistence_->Run("Write a doc and get an ack, not part of a target", [&] {
+    Document doc = CacheADocumentInTransaction();
+    // Mark it as eligible for GC, but this is after our upper bound for what we
+    // will collect.
+    MarkDocumentEligibleForGcInTransaction(doc.key());
+    // This should be retained, it's too new to get removed.
+    expected_retained.insert(doc.key());
+  });
+
+  // Finally, do the garbage collection, up to but not including the removal of
+  // middle_target
+  std::unordered_map<TargetId, QueryData> live_queries{
+      {oldest_target.target_id(), oldest_target}};
+
+  int queries_removed = RemoveTargets(upper_bound, live_queries);
+  ASSERT_EQ(1, queries_removed) << "Expected to remove newest target";
+  int docs_removed = RemoveOrphanedDocuments(upper_bound);
+  ASSERT_EQ(expected_removed.size(), docs_removed);
+  persistence_->Run("verify results", [&] {
+    for (const DocumentKey& key : expected_removed) {
+      ASSERT_EQ(document_cache_->Get(key), absl::nullopt)
+          << "Did not expect to find " << key.ToString().c_str()
+          << "in document cache";
+      ASSERT_FALSE(query_cache_->Contains(key))
+          << "Did not expect to find " << key.ToString().c_str()
+          << " in query_cache";
+      ExpectSentinelRemoved(key);
+    }
+    for (const DocumentKey& key : expected_retained) {
+      ASSERT_NE(document_cache_->Get(key), absl::nullopt)
+          << "Expected to find " << key.ToString().c_str()
+          << " in document cache";
+    }
+  });
+
+  persistence_->Shutdown();
+}
+
+TEST_P(LruGarbageCollectorTest, GetsSize) {
+  NewTestResources();
+
+  size_t initial_size = gc_->CalculateByteSize();
+
+  persistence_->Run("fill cache", [&] {
+    // Simulate a bunch of ack'd mutations
+    for (int i = 0; i < 50; i++) {
+      Document doc = CacheADocumentInTransaction();
+      MarkDocumentEligibleForGcInTransaction(doc.key());
+    }
+  });
+
+  size_t final_size = gc_->CalculateByteSize();
+  ASSERT_GT(final_size, initial_size);
+
+  persistence_->Shutdown();
+}
+
+TEST_P(LruGarbageCollectorTest, Disabled) {
+  LruParams params = LruParams::Disabled();
+  NewTestResources(params);
+
+  persistence_->Run("fill cache", [&] {
+    // Simulate a bunch of ack'd mutations
+    for (int i = 0; i < 500; i++) {
+      Document doc = CacheADocumentInTransaction();
+      MarkDocumentEligibleForGcInTransaction(doc.key());
+    }
+  });
+
+  LruResults results =
+      persistence_->Run("GC", [&] { return gc_->Collect({}); });
+  ASSERT_FALSE(results.did_run);
+
+  persistence_->Shutdown();
+}
+
+TEST_P(LruGarbageCollectorTest, CacheTooSmall) {
+  LruParams params = LruParams::Default();
+  NewTestResources(params);
+
+  persistence_->Run("fill cache", [&] {
+    // Simulate a bunch of ack'd mutations
```

Comment:
Ultranit: add full stop (applies throughout this function).

Reply:
Done.